### PR TITLE
Added WeighSize() to Softmax, Max_Pooling and Min_Pooling layers.

### PR DESCRIPTION
**Pull request by RishabhGarg108**
Issue #2590

### CODE REVIEW

**Comment by Aakash-kaushik** [comment]
Seems fine to me, can you fix the style check fails.

**Comment by RishabhGarg108** [comment]
@Aakash-kaushik , earlier the style fault was that two of the files had a space in the beginning of an empty line. I deleted that space, but it is still showing style errors. Can you please help me how to fix it.

**Comment by Aakash-kaushik** [comment]
> @Aakash-kaushik , earlier the style fault was that two of the files had a space in the beginning of an empty line. I deleted that space, but it is still showing style errors. Can you please help me how to fix it.

You have a extra space on the blank lines which are 162 in `max_pooling` and 143 in `mean_pooling`.

File: src/mlpack/methods/ann/layer/max_pooling.hpp [hunk 1]
```diff
@@ -159,7 +159,7 @@ class MaxPooling
   bool Deterministic() const { return deterministic; }
   //! Modify the value of the deterministic parameter.
   bool& Deterministic() { return deterministic; }
-  
+
```
Comment (RishabhGarg108) [inline src/mlpack/methods/ann/layer/max_pooling.hpp:162]: @Aakash-kaushik , have a look here. I removed that line with extra space and added a new blank line. Now, I am unable to understand why is it showing two errors here - one is "Weird number of spaces at line-start" and the other is "Line ends in whitespace".

**Comment by Aakash-kaushik** [comment]
extra-space

I cloned your branch and i still see an extra space on line 162 that is not on any other blank line, see line 165.

**Comment by RishabhGarg108** [comment]
@Aakash-kaushik , sorry for your trouble. It was actually some fault with vim. But, now I have corrected it and the style tests have passed too. A few tests are still running and I am positive that they will be successful.
Thanks for taking time to review this. It was my first PR and I know I didn't do much in it. But, it definitely helped me in understanding the workflow :D

**Comment by Aakash-kaushik** [comment]
> @Aakash-kaushik , sorry for your trouble. It was actually some fault with vim. But, now I have corrected it and the style tests have passed too. A few tests are still running and I am positive that they will be successful.
> Thanks for taking time to review this. It was my first PR and I know I didn't do much in it. But, it definitely helped me in understanding the workflow :D

Always happy to help and also the failed tests don't belong to the changes in this PR.
@zoq can you take a look ?

**Review by zoq — APPROVED** [review]
Looks good to me, thanks.

**Comment by RishabhGarg108** [comment]
This PR is approved. So, can some authorized person please merge this.

**Comment by zoq** [comment]
Thanks again for the contribution.

**Comment by RishabhGarg108** [comment]
Always happy to contibute :)

**Pull request merged**